vignettes/workflows.Rmd

```diff
@@ -0,0 +1,210 @@
+
+### Some comments on collaboration
+
+There are a number of features about the system here that might not be obvious at first, but that will aid collaboration.
+
+* Because each report exists in its own directory within `src/` we reduce the potential for conflicts.  In a collaborative analysis where multiple files are shared across analyses, it is easy to have conflicts.  `orderly` forces analyses into separate directories, which means most of the time it is trivial to merge even long-lived branches.  This is particularly important with relatively new users of git.
+* Because every report gets a unique identifier, it is straightforward to combine outputs that were run on different machines.  In the example above this is a centralised system (one canonical source, several destinations) but in theory more complicated distributed workflows are possible.
+* On the central server, the archive directory is an "append only" system; nothing is ever modified.  This seems wasteful at first as there can be lots of copies of a report, but disk space is cheaper than researcher time (and you can remove redundancy with `orderly::orderly_deduplicate`)
+
+## Group of researchers, sharing via server
+
+This is the model that we use within the [MRC Centre for Global Infectious Disease Analysis](https://www.imperial.ac.uk/mrc-global-infectious-disease-analysis), and it does require considerably more setup, but also more reliability and flexibility.
+
+As above, we use a single "source tree", kept on GitHub or similar.  However, someone in the group takes responsibility for setting up a copy of [OrderlyWeb](https://github.com/vimc/orderly-web), our open source, self-hosted orderly server.  The OrderlyWeb server software provides a number of things
+
+1. A persistent copy of R running `orderly` that can run reports as needed
+2. A web server that exposes a user-friendly web portal showing versions of orderly reports and an HTTP API (this application is written in [Kotlin](https://kotlinlang.org))
+3. A [reverse proxy](https://en.wikipedia.org/wiki/Reverse_proxy) used to secure the application with https
+
+This [diagram may help show how the pieces fit together](https://raw.githubusercontent.com/vimc/montagu-machine/master/docs/diagrams/OrderlyWeb%20architecture.png).
+
+If OrderlyWeb has been deployed onto `orderly.example.com`, then we might declare it in `orderly_config.yml`
+
+```yaml
+remote:
+  default:
+    driver: orderlyweb::orderlyweb_remote
+    args:
+      host: orderly.example.com
+      token: $GITHUB_TOKEN
+```
+
+Once this is done, both Alice and Bob can run reports *on* that server, for example
+
+```r
+orderly::orderly_run_remote("report-a", remote = "default")
+```
+
+**ADD A GIF**
+
+this will print to screen the log, just as if it was running locally, though it is in fact running on a fresh R session remotely.  After being run, this report is available to everyone and can be pulled as above
```

Comment:
as if it *were* running locally...

Reply:
Ah, the rare form of the English subjunctive